src/Components/NavBar/NavBar.tsx

```diff
@@ -86,7 +103,20 @@ export const NavBar: React.FC<NavBarProps> = () => {
 
             {isLoggedIn && (
               <>
-                <NavItem Menu={NotificationsMenu} Overlay={NotificationsBadge}>
+                <NavItem
+                  href="/works-for-you"
+                  Menu={NotificationsMenu}
+                  Overlay={NotificationsBadge}
+                  onClick={() => {
+                    tracking.trackEvent({
+                      flow: Schema.Flow.Header,
+                      context_module: Schema.ContextModule.Header,
+                      subject: Schema.Subject.NotificationBell,
+                      new_notification_count: cookie.get("notification-count"),
```

Comment:
Is there a way to decouple this and have the `<NavBar … />` call-site pass this in somehow? It feels a bit magical and unclear where/how this comes from.

Reply:
I agree this is a bit weird. Here's the flow: If the cookie has been set from a previous visit then it's possible to pluck this value from React's context, as the data is passed over from the server and injected into into it. But if it's a first visit, then the most direct way of getting this number is via cookie, because it's set right after relay finishes its fetch (which leaves some space open for a race condition should the user click, however unlikely, right when they land for the first time on the page.)

Reply:
I see. Would it be possible to insert the data into the Relay store, server-side, so that either way this can always pull from the store and either get the SSR value or the newly fetched value? That should also ensure it always automatically correctly reflects the count.

Reply:
Ah, that's interesting - let me think about this a bit... Seems possible!

Reply:
As we're currently deriving the notification count by the number of returned edges, would a good approach be to update Metaphysics the notifications type with a new `count` field, which I can then set via `setLinkedRecord` before passing the environment into `<SystemContext>`?